ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	pdnsSecondaryZones = "secondary_zones"
+	pdnsSZResolverID   = "resolver_id"
+	pdnsSZId           = "secondary_zone_id"
+	pdnsSZDescription  = "description"
+	pdnsSZZone         = "zone"
+	pdnsSZEnabled      = "enabled"
+	pdnsSZTransferFrom = "transfer_from"
+	pdnsSZCreatedOn    = "created_on"
+	pdnsSZModifiedOn   = "modified_on"
+	pdnsSZOffset       = "offset"
+	pdnsSZLimit        = "limit"
+)
+
+func DataSourceIBMPrivateDNSSecondaryZones() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMDNSSecondaryZonesRead,
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The GUID of the DNS Services instance.",
+			},
+			pdnsSZResolverID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a custom resolver.",
+			},
+			pdnsSecondaryZones: {
+				Type:        schema.TypeList,
+				Description: "List of Secondary Zones",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						pdnsSZId: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier of the Secondary Zone",
+						},
+						pdnsSZDescription: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Descriptive text of the secondary zone.",
+						},
+						pdnsSZZone: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the zone.",
+						},
+						pdnsSZEnabled: {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Enable/Disable the secondary zone.",
+						},
+						pdnsSZTransferFrom: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The addresses of DNS servers where the secondary zone data is transferred from.",
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						pdnsSZCreatedOn: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Time when a secondary zone is created",
+						},
+
+						pdnsSZModifiedOn: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The recent time when a secondary zone is modified",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMDNSSecondaryZonesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceID := d.Get(pdnsInstanceID).(string)
+	resolverID := d.Get(pdnsSZResolverID).(string)
+
+	opt := sess.NewListSecondaryZonesOptions(instanceID, resolverID)
+
+	result, resp, err := sess.ListSecondaryZonesWithContext(context, opt)
```

Comment:
Does this handles pagination if you have more than 50 or 100 secondary zones .... Will this list all or only the first page. If only first page then you need to add pagination support

Reply:
Maximum number of secondary zones allowed for now is 5. So this is not an issue for this release.